Sources/Widgets/SVG/SVGLandmarkRepresentation/index.js

```diff
@@ -14,48 +14,64 @@ function vtkSVGLandmarkRepresentation(publicAPI, model) {
   publicAPI.render = () => {
     const list = publicAPI.getRepresentationStates();
 
-    let index = 0;
     const coords = list.map((state) => state.getOrigin());
-    const texts = list.map((state) => {
-      const ret =
-        typeof state.getText !== 'undefined' ? state.getText() : `L${index}`;
-      index++;
-      return ret;
-    });
+    const texts = list.map((state, index) =>
+      state.getText ? state.getText() : `L${index}`
+    );
 
     return publicAPI.worldPointsToPixelSpace(coords).then((pixelSpace) => {
       const points2d = pixelSpace.coords;
       const winHeight = pixelSpace.windowSize[1];
 
       const root = createSvgElement('g');
-
       for (let i = 0; i < points2d.length; i++) {
         const xy = points2d[i];
-        const x = xy[0];
-        const y = winHeight - xy[1];
-
-        let circle = {};
-        if (model.showCircle === true) {
-          circle = createSvgElement('circle');
-          Object.keys(model.circleProps || {}).forEach((prop) =>
-            circle.setAttribute(prop, model.circleProps[prop])
-          );
-          circle.setAttribute('cx', x);
-          circle.setAttribute('cy', y);
-          root.appendChild(circle);
+        if (!Number.isNaN(xy[0]) && !Number.isNaN(xy[1])) {
+          const x = xy[0];
+          const y = winHeight - xy[1];
+
+          let circle = {};
+          if (model.showCircle === true) {
+            circle = createSvgElement('circle');
+            Object.keys(model.circleProps || {}).forEach((prop) =>
+              circle.setAttribute(prop, model.circleProps[prop])
+            );
+            circle.setAttribute('cx', x);
+            circle.setAttribute('cy', y);
+            root.appendChild(circle);
+          }
+          if (!texts[i]) {
+            texts[i] = '';
+          }
+          const splitText = texts[i].split('\n');
+          const newlineOffset =
+            model.fontProperties != null && model.fontProperties.fontSize
+              ? model.fontProperties.fontSize
+              : 15;
+          splitText.forEach((subText, j) => {
+            const text = createSvgElement('text');
+            Object.keys(model.textProps || {}).forEach((prop) => {
+              if (model.fromLineWidget === true && prop === 'dy') {
```

Comment:
I remember in a comment to rename `fromLineWidget`.
That variable should not define the context where SVGLandmarkRepresentation is instantiated from. Instead it should be explicit on the behavior to expect:
e.g. `newLines`.

I do not see a case where we do not want that variable to be set to True, do you ?

Reply:
for the normal behavior of SVGLandmarkRepresentation as it is in PolyLineWidget example the variable should not be evaluated to true